p2p/discovery/message.go

```diff
@@ -18,39 +20,189 @@ const (
 	NeighborsMsg MsgType = 5
 )
 
-type Ping struct {
+const (
+	VERSION uint = 1
+)
+
+type ping struct {
 	Version uint // TODO add version check
-	ID      NodeID
+	SelfID  NodeID
+
+	to NodeID
 }
 
-type Pong struct {
-	ID NodeID
+type pong struct {
+	SelfID NodeID
 }
 
-type FindNode struct {
-	Target NodeID
+type findNode struct {
+	SelfID  NodeID
+	QueryID NodeID
+
+	to NodeID
```

Comment:
SelfID is the node that launch the query, to is the target node, what's QueryID?

Reply:
queryID is which node we want to find